kernel/kernel-api/src/main/java/io/delta/kernel/data/FilteredColumnarBatch.java

```diff
@@ -37,10 +37,23 @@
 public class FilteredColumnarBatch {
   private final ColumnarBatch data;
   private final Optional<ColumnVector> selectionVector;
+  private final Optional<String> filePath;
+  // Optional pre-computed count of true rows in the selection vector
+  private Optional<Integer> numSelectedRows;
```

Comment:
nit: in general, for **internal code** when you have a detailed comment that explains what the variable is, try and just see if you can make the variable name more self-descriptive, and then we might not need the comment at all.

e.g. if this was `numPreComputedSelectedRows` -- i don't think we would need the comment then at all?